src/main/java/bio/terra/cloudfunctions/common/CloudStorageEventHarness.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.cloudfunctions.common;
+
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.gson.internal.LinkedTreeMap;
+import java.util.logging.Logger;
+
+/**
+ * This class is an abstract representation of a Function triggered by a Google Cloud Storage Event.
+ * The function casts the Google Cloud Storage Event into a Map of key-value pairs.
+ *
+ * <p>This class contains only framework-specific logic to receive and parse Google Storage Object
+ * Event messages.
+ *
+ * <p>Sub-classes of this class can be integrated with a DI framework to deploy Cloud Functions like
+ * a Service.
+ */
+public abstract class CloudStorageEventHarness implements BackgroundFunction<LinkedTreeMap<?, ?>> {
```

Comment:
This class is coupled to Google's API and I think we should include Google in the name of the class to be extra clear about this.

Reply:
I think this class should implement `RawBackgroundFunction` and store the event as a string and then deserialize from that string in `getEvent` What do we gain by using the LinkedTreeMap to hold the event data?  Here's a suggestion for the class: 
```suggestion
public abstract class CloudStorageEventHarness implements RawBackgroundFunction {
  private static final Logger logger = Logger.getLogger(CloudStorageEventHarness.class.getName());

  private Context context;
  private String event;


  @Override
  public void accept(String event, Context context) throws Exception {
    this.event = event;
    this.context = context;
    doAccept();
  }


  /** @return Google Cloud Storage Event Context */
  public Context getContext() {
    return context;
  }

  /**
   * @param classOfT generic type of event pojo
   * @param <T> generic event pojo
   * @return an event pojo of generic type T matching the provider's specification of the event. *
   *     (e.g. GCSEvent is a pojo of GCS StorageObjectData event). Other providers have equivalent
   *     events * such as AWS S3Event and Azure BlobTrigger. Different providers use slightly
   *     different naming and * syntax for the event trigger (e.g. accept for GCS, handleRequest for
   *     S3, run for Azure Blob * Storage)
   */
  public <T> T getEvent(Class<T> classOfT) {
    try {
      return GsonWrapper.convertFromClass(this.event, classOfT);
    } catch (JsonSyntaxException e) {
      logger.log(Level.SEVERE, "Could not convert from JSON", e);
      throw e;
    }
  }

  /**
   * Subclasses of this class must implement this method to kick off application-specific functions
   * downstream .
   *
   * @throws Exception when something wrong happened
   */
  public abstract void doAccept() throws Exception;
}
```

Reply:
Renamed to `GoogleCloudStorageEventHarness`.